infra/aws/terraform/prow-build-cluster/docs/EKSUpgradeRunbook.md

```diff
@@ -0,0 +1,55 @@
+# EKS Kubernetes Version Upgrade
+
+## Control Plane Upgrade
+
+To upgrade the Kubernetes version, you need to update the `cluster_version` variable on the cluster's tfvar files:
+
+```diff
+## canary cluster:
+#infra/aws/terraform/prow-build-cluster/terraform.canary.tfvars
+
+## prod cluster:
+#infra/aws/terraform/prow-build-cluster/terraform.prod.tfvars
+
+--- a/infra/aws/terraform/prow-build-cluster/terraform.canary.tfvars
++++ b/infra/aws/terraform/prow-build-cluster/terraform.canary.tfvars
+@@ -31,13 +31,13 @@ eks_cluster_admins = [
+ ]
+ 
+ cluster_name               = "prow-canary-cluster"
+-cluster_version            = "1.28"
++cluster_version            = "1.29"
+```
+
+After this change, you need to run `make plan` and `make apply` commands. (Checkout [IaC Document](./IaC.md) for more information).
+
+
+## Worker Nodes Upgrade
+
+### Karpenter
+
+The current configuraiotn for Karpenter NodePool uses `amiFamily` setting. As a result, any new nodes provisioned after the cluster upgrade will automatically use the latest AMI for the new Kubernetes version.
+
+If you need to upgrade already provisioned nodes immediately, you should manually cordon, drain, and terminate those nodes to replace them with new nodes running the updated Kubernetes version.
```

Comment:
We should document this process if possible. I think it should be enough to delete the NodeClaim object for the node.

Reply:
I tested on the canary cluster, and draining the node is sufficient. Karpenter removed the node automatically due to emptiness. Updated the doc accordingly, wdyt?